doc_files/algorithms.md

```diff
@@ -39,8 +39,34 @@ In A/B or A/B/n testing, the "optimality" of a version relates to maximizing the
 
 ## 3. Posterior Bayesian Routing (`pbr`)
 
-One of our novel algorithms. Coming soon to iter8...
+```yaml
+interval: # (time; e.g., 30s)
+maxIterations: # (integer; e.g., 1000)
+maxTrafficPercentage: # 80 (percentage; e.g., 90)
+confidence: # (float; e.g 0.95)
+onSuccess: # (string enum; possible values are: "candidate", "baseline", "both")
+```
+
+This algorithm like the epsilon-greedy strategy described above, can be applied to canary releases as well as A/B or A/B/n testing scenarios. The goal of this strategy is similar to that of the one above. It is to shift traffic to optimal versions of a micro service not just based on their reward attribute but also based on other attributes which need to satisfy their respective feasibility constraints (i.e. user-defined success criteria).
+
+Here, the reward and feasibility constraints are viewed in the form of beta/normal distributions which are sampled from while calculating traffic split between the different versions.
+
+This algorithm is novel in that, for each iteration the increase in traffic to the version that performs best (i.e satisfies all user-defined success criteria as well as obtains the maximum reward) is dependent on the load already generated to the version. This ensures that the traffic split generated has high confidence.
+
+Another useful point to note is that Bayesian Routing algorithms can make use of historical data to perform analysis of different micro service versions. Through our experiments, we have found that this algorithm converges much quicker than the epsilon-greedy and check-and-increment strategies to the most feasible micro service version.
 
 ## 4. Optimistic Bayesian Routing (`obr`)
 
-One of our novel algorithms. Coming soon to iter8...
+```yaml
+interval: # (time; e.g., 30s)
+maxIterations: # (integer; e.g., 1000)
+maxTrafficPercentage: # 80 (percentage; e.g., 90)
+confidence: # (float; e.g 0.95)
+onSuccess: # (string enum; possible values are: "candidate", "baseline", "both")
+```
+
+This algorithm is almost exactly similar to the one above. It can be adopted to A/B or A/B/n scenarios. It aims to shift traffic to a feasible micro service version based on their reward attibute and different user-defined success criteria.
```

Comment:
If it is almost exactly the same, can we drop most of this paragraph and the next and go straight to the difference?